components/datetime/src/arithmetic.rs

```diff
@@ -0,0 +1,91 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/master/LICENSE ).
+
+//! Assorted functions to help with date calculations.
+
+use crate::date::{Era, IsoWeekday, Year};
+use crate::pattern::{Pattern, TimeGranularity};
+
+use tinystr::tinystr8;
+
+pub fn iso_year_to_gregorian(iso_year: i32) -> Year {
+    if iso_year > 0 {
+        Year {
+            era: Era(tinystr8!("ce")),
+            number: iso_year,
+            related_iso: iso_year,
+        }
+    } else {
+        Year {
+            era: Era(tinystr8!("bce")),
+            number: -iso_year + 1,
+            related_iso: iso_year,
+        }
+    }
+}
+
+#[test]
+fn test_iso_year_to_gregorian() {
+    assert_eq!(
+        iso_year_to_gregorian(2020),
+        Year {
+            era: Era(tinystr8!("ce")),
+            number: 2020,
+            related_iso: 2020,
+        }
+    );
+    assert_eq!(
+        iso_year_to_gregorian(1),
+        Year {
+            era: Era(tinystr8!("ce")),
+            number: 1,
+            related_iso: 1,
+        }
+    );
+    assert_eq!(
+        iso_year_to_gregorian(0),
+        Year {
+            era: Era(tinystr8!("bce")),
+            number: 1,
+            related_iso: 0,
+        }
+    );
+    assert_eq!(
+        iso_year_to_gregorian(-1),
+        Year {
+            era: Era(tinystr8!("bce")),
+            number: 2,
+            related_iso: -1,
+        }
+    );
+}
+
+/// Temporary simplified function to get the day of the week
+/// month and day are both zero-indexed.
+pub fn iso_date_to_weekday(year: i32, month: usize, day: usize) -> IsoWeekday {
```

Comment:
Good point.  Since I removed the `< 12` requirement on `month`, there are multiple places where we could trigger an out-of-range error.  This is one; another is in `get_symbol_for_month`.

For now, I'll add `debug_assert!` in these places, but we don't actually enforce this anywhere anymore.  We should follow up with this in https://github.com/unicode-org/icu4x/issues/493.